File: onlineshop4dvds_api/src/products/dtos/product-detail.dto.ts
export interface ProductDetail {
    id: number;
    title: string;
    price: number;
    description: string;
    genres: string[];
    ratings: number | null;
    numbersOfReviews: number;
    images: string[];
    yearReleased: number;
    stock: number;
}

export interface AlbumProductDetailDto extends ProductDetail {
    artist: string;
    artistAvatar: string;
    lengthInSeconds: number;
}

export interface GameProductDetailDto extends ProductDetail {
    console: string;
}

export interface MovieProductDetailDto extends ProductDetail {
    imdbRatings: number;
}

File: onlineshop4dvds_api/DTOs/ProductDetailDto.cs
namespace OnlineShop4DVDS.DTOs;

public class ProductDetailDto
{
    public required int Id { get; set; }
    public required string Title { get; set; }
    public required decimal Price { get; set; }
    public required string Description { get; set; }
    public required ICollection<string> Genres { get; set; }
    public required float Ratings { get; set; }
    public required int NumbersOfReviews { get; set; }
    public required int Stock { get; set; }
}
